packages/grafana-ui/src/components/Logs/LogRows.tsx

```diff
@@ -137,7 +137,7 @@ class UnThemedLogRows extends PureComponent<Props, State> {
             {hasData &&
               firstRows.map((row, index) => (
                 <LogRow
-                  key={row.uid}
+                  key={`${row.uid}-${index}`}
```

Comment:
Uid is created by hashing all provided logs information. However, we might have multiple identical logs.Therefore we need to add index here. This pattern is used at other places in code base as well.

Reply:
hmm what I don't like with this is that we have a property uid, which usually means unique identifier and that right away say, this may not be unique. From the meaning of that property, the data source should provide it and if it doesn't we do some fall back to something when processing the logs (I think we already just add an index there or something). So this does not feel right I think we should either abandon creation of uid in the Loki DS because it does not actually provide uids or maybe do an uuid4 or something like that in the datasource.